e2e/single-cluster/imagescan_test.go

```diff
@@ -92,3 +68,211 @@ var _ = Describe("Image Scan", Label("infra-setup"), func() {
+
+	AfterEach(func() {
+		_, _ = k.Delete("gitrepo", "imagescan")
+		_, _ = k.Delete("secret", "git-auth")
+	})
+	When("deploy imagescan setup with pre-release images", func() {
+		BeforeEach(func() {
+			assetdir = "imagescan/pre-releases-ok"
+			tmpRepoDir = GinkgoT().TempDir()
+			image, imageTag = initRegistryWithImageAndTag("k8s.gcr.io/pause", "0.0.0-40")
+			applyTemplateValues(assetdir, tmpRepoDir, image, imageTag)
+		})
+		It("updates the docker reference", func() {
+			By("checking the deployment exists")
+			Eventually(func() string {
+				out, _ := k.Namespace("default").Get("pods")
+				return out
+			}).Should(ContainSubstring("pause-prerelease-"))
+
+			By("checking the bundle has the original image tag")
+			Eventually(func() string {
+				out, _ := k.Get("bundles", "imagescan-examples", "-o", "yaml")
+				return out
+			}).Should(
+				MatchRegexp(fmt.Sprintf(`image: %s # {"\$imagescan": "test-scan"}`, imageTag)))
+
+			By("pushing a new tag to the registry and checking the new image tag is found in the bundle")
+			newTag := "0.0.0-50"
+			imageTag = tagAndPushImage("k8s.gcr.io/pause", image, newTag)
+			Eventually(func() string {
+				out, _ := k.Get("bundles", "imagescan-examples", "-o", "yaml")
+				return out
+			}).Should(
+				MatchRegexp(fmt.Sprintf(`image: %s # {"\$imagescan": "test-scan"}`, imageTag)))
+
+			By("checking that the new tag is pushed in the git repository")
+			err := gh.CheckoutRemote(repository, "imagescan")
+			Expect(err).NotTo(HaveOccurred())
+			basedir := filepath.Join(clonedir, "examples")
+			b, err := os.ReadFile(filepath.Join(basedir, "deployment.yaml"))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(b)).Should(ContainSubstring(newTag))
+		})
+	})
+
+	// this test didn't pass before adding fix for issue #2096
+	When("deploy imagescan setup with pre-release images and * semver range", func() {
+		BeforeEach(func() {
+			assetdir = "imagescan/pre-releases-ignored"
+			tmpRepoDir = GinkgoT().TempDir()
+			image, imageTag = initRegistryWithImageAndTag("k8s.gcr.io/pause", "0.0.0-40")
+			applyTemplateValues(assetdir, tmpRepoDir, image, imageTag)
+		})
+		It("updates the image reference", func() {
+			By("checking the deployment exists")
+			Eventually(func() string {
+				out, _ := k.Namespace("default").Get("pods")
+				return out
+			}).Should(ContainSubstring("pause-prerelease-"))
+
+			By("checking the bundle has the original image tag")
+			Eventually(func() string {
+				out, _ := k.Get("bundles", "imagescan-examples", "-o", "yaml")
+				return out
+			}).Should(
+				MatchRegexp(fmt.Sprintf(`image: %s # {"\$imagescan": "test-scan"}`, imageTag)))
+
+			By("pushing a new tag to the registry and checking the fleet controller does not crash")
+			// store number of fleet controller restarts to compare later
+			fleetControllerRestarts := getFleetControllerRestarts(k)
+			newTag := "0.0.0-50"
+			previousImageTag := imageTag
+			imageTag = tagAndPushImage("k8s.gcr.io/pause", image, newTag)
```

Comment:
This doesn't seem to be used anywhere, so perhaps we could `s/previousImageTag/imageTag/`.

Reply:
`previousImageTag` is checked here
```go
Eventually(func() string {
				out, _ := k.Get("bundles", "imagescan-examples", "-o", "yaml")
				return out
			}).Should(
				MatchRegexp(fmt.Sprintf(`image: %s # {"\$imagescan": "test-scan"}`, previousImageTag)))
``` 
`imageTag` is no longer used, so technically we could avoid storing `imageTag` and use the previous value.

I did it this way to make it clear that `previousImageTag` refers to `0.0.0-40` and that `imageTag` refers to the latest pushed tag (`0.0.0-50` in this case)

Reply:
Yes, sorry, what I meant is that `imageTag` is computed with the new tag but not used.
On the other hand, I like the extra clarity provided by `previousImageTag` :+1: